common/admin/osd/osd_blocklist.go

```diff
@@ -0,0 +1,183 @@
+//go:build !octopus && ceph_preview
+
+package osd
+
+import (
+	"fmt"
+	"net"
+	"regexp"
+
+	"github.com/ceph/go-ceph/internal/commands"
+)
+
+type ipList struct {
+	IPAddr string `json:"addr"`
+	Until  string `json:"until"`
+}
+
+type networkList struct {
+	Network string `json:"range"`
+	Until   string `json:"until"`
+}
+
+// Blocklist contains the address and expire value for a blocklist entry.
+type Blocklist struct {
+	Addr  string
+	Until string
+}
+
+func parseBlocklist(res response) (*[]Blocklist, error) {
+	var ip []ipList
+	var nw []networkList
+	var bl []Blocklist
+
+	matches := regexp.MustCompile(`\[.*?\]`).FindAllSubmatch(res.Body(), -1)
```

Comment:
OK, I think I still think the Decoder approach I mentioned over chat would be a little bit more general but this is fine with me. I don't expect this ceph command to return nested lists and if they started to I hope it would be *after* they fix the JSON output generally.

Reply:
OK, thanks, I appreciate the extra effort.